subprojects/core/src/main/java/org/gradle/initialization/buildsrc/BuildSourceBuilder.java

```diff
@@ -93,14 +105,29 @@ public String getBuildPath() {
     }
 
     private ClassPath buildBuildSrc(StartParameter startParameter) {
-        BuildController buildController = createBuildController(startParameter);
+        final PersistentCache buildSrcCache = createCache(startParameter);
         try {
-            return new BuildSrcUpdateFactory(buildController, buildSrcBuildListenerFactory).create();
+            BuildController buildController = createBuildController(startParameter);
+            try {
+                return buildSrcCache.useCache(new BuildSrcUpdateFactory(buildController, buildSrcBuildListenerFactory, cachedClasspathTransformer));
+            } finally {
+                buildController.stop();
+            }
         } finally {
-            buildController.stop();
+            buildSrcCache.close();
         }
     }
 
+    PersistentCache createCache(StartParameter startParameter) {
+        return cacheRepository
+            .cache(new File(startParameter.getCurrentDir(), ".gradle/noVersion/buildSrc"))
```

Comment:
⭕️ Why is there the word "noVersion" here when further down we put the gradle.version into a properties file?

Reply:
The location of the cache is always the same. The cache only gets invalidated when the properties are different. Locking still works.
Some time ago we used that information to decide whether to run `clean` on buildSrc or not (when the Gradle version changed).